openfl-tutorials/interactive_api/HPU/PyTorch_Kvasir_UNet/PyTorch_Kvasir_UNet.ipynb

```diff
@@ -0,0 +1,645 @@
+    "fl_experiment.start(model_provider=MI, \n",
+    "                    task_keeper=TI,\n",
+    "                    data_loader=fed_dataset,\n",
+    "                    rounds_to_train=2,\n",
+    "                    opt_treatment='CONTINUE_GLOBAL',\n",
+    "                    device_assignment_policy='CUDA_PREFERRED')\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f1543a36",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# If user want to stop IPython session, then reconnect and check how experiment is going \n",
+    "# fl_experiment.restore_experiment_state(MI)\n",
+    "\n",
+    "fl_experiment.stream_metrics()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8c30b301",
+   "metadata": {},
+   "source": [
+    "## Now we validate the best model!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "55acff59",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Now we validate the best model!\\n\")\n",
+    "best_model = fl_experiment.get_best_model()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9479fb7f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # We remove exremove_experiment_datamove_experiment_datamove_experiment_datariment data from director\n",
+    "fl_experiment.remove_experiment_data()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "75c8aeab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "best_model.inc.conv[0].weight\n",
+    "# model_unet.inc.conv[0].weight"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a2acb7e6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Validating initial model\n",
+    "validate(initial_model, fed_dataset.get_valid_loader(), 'hpu')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c12ca93f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Validating trained model\n",
+    "validate(best_model, fed_dataset.get_valid_loader(), 'hpu')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e6734f6",
+   "metadata": {},
+   "source": [
+    "## We can tune model further!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3940e75e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# MI = ModelInterface(model=best_model, optimizer=optimizer_adam, framework_plugin=framework_adapter)\n",
+    "fl_experiment.start(model_provider=MI, task_keeper=TI, data_loader=fed_dataset, rounds_to_train=2, \\\n",
+    "                              opt_treatment='CONTINUE_GLOBAL')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a9490ed9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fl_experiment.stream_metrics()"
```

Comment:
Added this to avoid grpc connection error while running the federation for the 2nd time

Reply:
What does the line do? Add some documentation (?)

Reply:
Added the documentation